src/Form/AssignUserForm.php

```diff
@@ -0,0 +1,268 @@
+      $options = $this->getFormOptions($scheme);
+      $role_sections = $this->roleSectionStorage->getRoleSections($scheme, $user);
+      foreach ($options as $value => $label) {
+        if (in_array($value, $role_sections)) {
+          $options[$value] = '<strong>' . $label . ' * </strong>';
+        }
+      }
+      if (!empty($options)) {
+        $form[$scheme->id()] = [
+          '#type' => 'fieldset',
+          '#collapsible' => TRUE,
+          '#collapsed' => FALSE,
+          '#title' => $scheme->getPluralLabel(),
+        ];
+        $form[$scheme->id()]['active_' . $scheme->id()] = [
+          '#type' => 'checkboxes',
+          '#title' => $this->t('Assigned sections'),
+          '#options' => $options,
+          '#default_value' => $user_sections,
+          '#description' => $this->t('Sections assigned by role are <strong>emphasized</strong> and marked with an * but not selected unless they are also assigned directly to the user. They need not be selected. Access granted by role cannot be revoked from this form.'),
+        ];
+        $form[$scheme->id()]['scheme_' . $scheme->id()] = [
+          '#type' => 'value',
+          '#value' => $scheme,
+        ];
+        $form_enabled = TRUE;
+        $active_schemes[] = $scheme->id();
+      }
+    }
+    if ($form_enabled) {
+      $form['schemes'] = [
+        '#type' => 'value',
+        '#value' => $active_schemes,
+      ];
+      $form['actions'] = [
+        '#type' => 'actions',
+        'submit' => [
+          '#type' => 'submit',
+          '#name' => 'save',
+          '#value' => $this->t('Save'),
+        ],
+      ];
+    }
+    else {
+      $form['help'] = [
+        '#markup' => $this->t('You do not have permission to manage any assignments.'),
+      ];
+    }
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $values = $form_state->getValues();
+    $items = [];
+    foreach ($values['schemes'] as $id) {
+      $items[$id]['scheme'] = $values['scheme_' . $id];
+      $items[$id]['selections'] = $values['active_' . $id];
+    }
+    foreach ($items as $item) {
+      // Add sections.
+      $sections = array_filter($item['selections'], function($val) {
+        return !empty($val);
+      });
+      $sections = array_keys($sections);
+      $this->userSectionStorage->addUser($item['scheme'], $this->user, $sections);
+
+      // Remove sections.
+      $remove_sections = array_keys(array_filter($item['selections'], function($val) {
+        return empty($val);
+      }));
+      $this->userSectionStorage->removeUser($item['scheme'], $this->user, $remove_sections);
+    }
+  }
+
+  /**
+   * Checks access for the form.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function checkAccess() {
+    $permissions = ['assign workbench access', 'assign selected workbench access'];
```

Comment:
I think routing.yml supports a `+` format for multiple permissions

Reply:
It does! fixed.